rdt-open-reader/src/covidIndonesia/java/io/ona/rdt/widget/CovidRepeatingGroupFactory.java

```diff
@@ -41,9 +47,36 @@
         referenceEditText.setOnEditorActionListener(null);
         referenceEditText.setOnFocusChangeListener((v, hasFocus) -> {
             if (!hasFocus) {
+                updateCountObject(widgetArgs);
                 addOnDoneAction(referenceEditText, doneButton, widgetArgs);
             }
         });
+
+        updateCountObject(widgetArgs);
         return views;
     }
+
+    private void updateCountObject(WidgetArgs widgetArgs) {
+        try {
+            JSONArray stepFields = getStepFields(getJsonApi(widgetArgs).getStep(widgetArgs.getStepName()));
+            for (int i = 0; i < stepFields.length(); i++) {
+                JSONObject stepObject = stepFields.getJSONObject(i);
+                if ((widgetArgs.getJsonObject().getString(JsonFormConstants.KEY) + "_count").equals(stepObject.getString("key"))
+                        && StringUtils.isBlank(stepObject.optString(JsonFormConstants.VALUE, ""))) {
+                    stepObject.put(JsonFormConstants.VALUE, "0");
+                }
+            }
+        } catch (JSONException e) {
+            Timber.e(e);
+        }
+    }
```

Comment:
Logic about defaulting to "0" should ideally move to native forms where it is much easier to make the change i.e. set the value attribute to "0" for a new `repeatingGroupCountObj` [here](https://github.com/OpenSRP/opensrp-client-native-form/blob/c09bc4dd319007b520aa8b0159e8141c2cd21112/android-json-form-wizard/src/main/java/com/vijay/jsonwizard/widgets/RepeatingGroupFactory.java#L138).

Reply:
yes, it would be better if we move this logic to the native form library so can I do that in the native form instead of to make the change in RDT.